### PR TITLE
Add login screen, deprecate old token login

### PR DESCRIPTION
**Pull request by joeyyang**
@tstirrat please review and merge! @berlinjohnson if you wouldn't mind taking a look at the visual design, that'd also be wonderful! 

This PR deprecates the old token login and changes the login mechanism to email / password.

### CODE REVIEW

**Comment by tstirrat** [comment]
A couple of things:

1. Center the app selector dropdown and green button, it feels very lop-sided.

1. The error for invalid email is inline in the field, but the error for failed login credentials failed is an alert bar at the top. In my particular use-case I had scrolled slightly down on the login form  before submitting and so the login error wasn't visible. 

I feel like an "invalid login credentials error" error could go down near the login button.

1. When selecting an app that has collaborators (note that I am the owner on some and not on others), it appears to log me out, I get thrown back to the login page with an error.

**Comment by joeyyang** [comment]
@tstirrat thanks; back to you!

**Comment by joeyyang** [comment]
@tstirrat let me know when you get a chance to review this — I'm going to take a look at https://github.com/firebase/vulcan/pull/41 and https://github.com/firebase/vulcan/pull/40 today as well

**Comment by tstirrat** [comment]
I can't build/test this until we merge in #40

**Comment by joeyyang** [comment]
oh oh ok i'll get on that then

**Comment by joeyyang** [comment]
@tstirrat back to you!

**Comment by tstirrat** [comment]
Still getting "You do not have permission to view this Firebase." on a firebase app that I own (but has one other collaborator)

However, another app has the same configuration yet I am able to get in.

File: app/js/components/form-selectfirebase.jsx [hunk 1]
```diff
@@ -0,0 +1,223 @@
+/** @jsx React.DOM */
+var React = require('react/addons');
+var AppMixins = require('./mixins');
+var $ = require('jquery');
+
+/*
+* LOGIN FORM COMPONENT
```
Comment (tstirrat) [inline app/js/components/form-selectfirebase.jsx]: incorrect component description

File: app/js/components/form-selectfirebase.jsx [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+      // POST RESULTS BACK TO CONNECTION
+      self.setState({
+        firebases: firebasesAndSecrets,
+        selectedNamespace: firstNamespaceListed
+      });
+    };
+
+
+    // KICK OFF REQUESTS AND ULTIMATELY POST MESSAGE BACK TO APP CODE WITH FIREBASES / SECRETS
+    $.ajax({
```
Comment (tstirrat) [inline app/js/components/form-selectfirebase.jsx]: This component is called `form-selectfirebase` but turns out to be more complex given that it fetches and parses the list of firebases. Can we move this fetching logic outside into the parent context with the other complex app logic and simply pass down the list, making this a dumb component?

File: app/js/components/container.jsx [hunk 3]
```diff
@@ -1,15 +1,20 @@
 /** @jsx React.DOM */
+var $ = require('jquery');
```
Comment (tstirrat) [inline app/js/components/container.jsx]: These import from NPM but the packages are not installed, resulting in:

```
Error: Cannot find module 'firebase' from '/Users/tstirrat/dev/vulcan/app/js/components'

Error: Cannot find module 'jquery' from '/Users/tstirrat/dev/vulcan/app/js/components'
```

Once we import from npm, we can remove from `bower.json` (and `index.html`).

`bower.json` becomes pretty useless, so we should probably just remove that completely.

**Comment by tstirrat** [comment]
Navigating to a path in the vulcan url bar works, but throws this error in the console:
```
Invariant Violation: replaceState(...): Can only update a mounted or mounting component.
```

**Comment by tstirrat** [comment]
@joeyyang back to you :car:

File: app/js/components/form-selectfirebase.jsx (hunk 1 above)
Comment (joeyyang) [inline app/js/components/form-selectfirebase.jsx]: lol smh

File: app/js/components/form-selectfirebase.jsx (hunk 2 above)
Comment (joeyyang) [inline app/js/components/form-selectfirebase.jsx]: that makes sense

**Comment by joeyyang** [comment]
@tstirrat back to you!

File: chrome-extension/manifest.json [hunk 4]
```diff
@@ -5,14 +5,15 @@
...
+  "version": "1.4.0",
   "icons": {
     "16":  "images/icon16.png",
     "48":  "images/icon48.png",
     "128": "images/icon128.png"
   },
   "permissions": [
-    "storage"
+    "storage",
+    "https://*.firebase.com"
```
Comment (tstirrat) [inline chrome-extension/manifest.json]: This is causing an error too:
```
Permission 'https://*.firebase.com' is unknown or URL pattern is malformed.
```

File: app/js/components/form-selectfirebase.jsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+          <ul className={pclass(formClasses)}>
+            <li>
+              <label className={pclass('form-select-labels')} for='firebase-picker'>Select a Firebase app to view:</label>
+              <select className={pclass('form-select-controls')}
+                      id='firebase-picker'
+                      name='firebase-picker'
+                      ref='firebasePicker'
+                      onChange={this.updateSelectedFirebase}>
+                {this.props.firebases.map(function(firebase){
+                  return <option value={firebase.namespace}
```
Comment (tstirrat) [inline app/js/components/form-selectfirebase.jsx]: put a `key` on these children to stop the react warning

File: app/js/components/container.jsx [hunk 6]
```diff
@@ -388,6 +438,27 @@ module.exports = React.createClass({
       this.setState(newState);
     }.bind(this);
 
+    var selectFirebase = function(namespace, secret) {
+      var self = this;
+      var url = 'https://' + namespace + '.firebaseio.com';
+      var ref = new Firebase(url);
```
Comment (tstirrat) [inline app/js/components/container.jsx]: If you try to view app before selecting something in the firebase selector  (even though it appears your first one is selected) you get this:

```
Cannot parse Firebase url. Please use https://<YOUR FIREBASE>.firebaseio.com
```

**Comment by tstirrat** [comment]
@joeyyang :tennis:

**Comment by joeyyang** [comment]
@tstirrat :tennis:

**Comment by berlinjohnson** [comment]
:tennis: :tennis: :tennis: :tennis: :tennis: :tennis: :tennis: :tennis: :tennis:

**Comment by berlinjohnson** [comment]
:baseball: :basketball:

**Pull request merged**